src/topology/algebra/uniform_ring.lean

```diff
@@ -150,8 +151,45 @@ instance : comm_ring (completion R) :=
       (assume a b, by rw [← coe_mul, ← coe_mul, mul_comm]),
  ..completion.ring }
 
-end uniform_space.completion
+namespace ring_completion
+
+instance : algebra R (completion R) :=
+(uniform_space.completion.coe_ring_hom : R →+* (completion R)).to_algebra
+
+lemma algebra_smul_eq (r : R) (x : (completion R)) :
+  r • x = (r : completion R) * x :=
+rfl
+
+section algebra
+variables (S : Type*) [comm_semiring S] [algebra S R]
+
+instance algebra' : algebra S (completion R) :=
+((uniform_space.completion.coe_ring_hom : R →+* completion R).comp (algebra_map S R)).to_algebra
+
+@[simp] lemma algebra'_smul_eq (s : S) (x : completion R) :
+  s • x = (algebra_map S R s : completion R) * x :=
+rfl
+
+lemma coe_algebra_map (s : S) :
+  (algebra_map S R s : completion R) = algebra_map S (completion R) s :=
+rfl
 
+lemma coe_smul  (s : S) (r : R) :
+  ((s • r : R) : completion R) = (s • r : completion R) :=
+by rw [algebra'_smul_eq, ← coe_mul, algebra.smul_def]
+
+instance is_scalar_tower : is_scalar_tower S R (completion R) :=
+⟨λ r k x,
+begin
+  simp only [←mul_assoc, algebra.smul_def, map_mul],
+  congr,
+end⟩
```

Comment:
I think once you make the changes above you'll find that these two declarations already exist.